docs/import/color_function_util.js

```diff
@@ -25,22 +179,136 @@ function withColor(td, layer, property, index) {
+    return;
+  }
+  const colorFunctionDiv = $('#color-fxn-editor');
+  if (colorFunctionDiv.is(':visible') && td === globalTd) {
+    colorFunctionDiv.hide();
+    return;
+  }
+  colorFunctionDiv.show();
+  if (td === globalTd) {
+    return;
+  }
+  globalTd = td;
+  $('#' + colorStyleTypeStrings.get(type) + '-radio')
+      .prop('checked', true)
+      .trigger('change');
+}
+
+/**
+ * Switches the schema of {@link globalTd} to the given type.
+ * @param {enum} type
+ */
+function switchSchema(type) {
+  const colorFunction = getColorFunction();
+  $('.color-type-div').hide();
+  switch (type) {
+    case ColorStyle.SINGLE:
+      $('#single-color-radio').prop('checked', true);
+      $('#single-color-picker').val(colorFunction['color']);
+      $('#single').show();
+      break;
+    case ColorStyle.CONTINUOUS:
+      $('#continuous-radio').prop('checked', true);
+      $('#continuous-color-picker').val(colorFunction['color']);
+      populatePropertyPicker($('#continuous-property-picker'));
+      $('#continuous').show();
+      break;
+    case ColorStyle.DISCRETE:
+      $('#discrete-radio').prop('checked', true);
+      const propertyPicker = $('#discrete-property-picker');
+      populatePropertyPicker(propertyPicker);
+      if (propertyPicker.val()) populateDiscreteColorPickers();
+      $('#discrete').show();
+      break;
+  }
+  populateColorTd(type === ColorStyle.DISCRETE);
+  colorFunction['current-style'] = type;
+  updateLayersInFirestore();
+}
+
+/**
+ * Updates the given select element with the properties of the earth engine
+ * asset related to {@link globalTd}.
+ * @param {JQuery<HTMLElement>} picker
+ */
+function populatePropertyPicker(picker) {
+  picker.empty();
+  const colorFunction = getColorFunction();
+  const properties = colorFunction['columns'];
+  const asOptions = [];
+  Object.keys(properties).forEach((key) => {
+    asOptions.push($(document.createElement('option')).val(key).text(key));
+  });
+  picker.append(asOptions).val(colorFunction['field']);
+}
+
+/**
+ * Updates the list of color pickers for the discrete schema to have one for
+ * each property of the earth engine asset related to {@link globalTd}.
+ *
+ * We add a piece of data to each of these pickers so they know what property
+ * they're attached to.
+ */
+function populateDiscreteColorPickers() {
+  const pickerList = $('#discrete-color-pickers').empty();
+  const colorFunction = getColorFunction();
+  const values =
+      colorFunction['columns'][$('#discrete-property-picker').val()]['values'];
+  const asColorPickers = [];
+  const colors = getColorFunction()['colors'];
+  for (const value of values) {
+    const li = $(document.createElement('li'));
+    li.append($(document.createElement('label')).text(value + ': '));
+    const colorPicker =
+        createColorPicker()
+            .on('change', (event) => setDiscreteColor($(event.target)))
+            .data('value', value)
+            .val(colors ? colors[value] : null);
```

Comment:
the colors map of value -> color (not to be confused with color, sorry probably not a great choice on my part) doesn't exist for an asset until you actually change one of the color pickers in the discrete div (see setDiscreteColor). Sort of nice because hopefully it will never exist for some assets that don't care about it.

Reply:
As discussed offline, I'm in favor of standardizing the sync with Firestore so that the page view always corresponds to what's in Firestore. Add TODOs here and elsewhere to get rid of these guards once data is eagerly populated?

Reply:
Done.